src/main.rs

```diff
@@ -96,8 +129,159 @@ fn main() {
                 port,
             );
 
-            // TODO: create client instance
+            // create client instance and connect to server
+            let mut connection = TcpStream::connect((ip, port)).unwrap();
+
+            // add CTRL+C system hook, so that connection partner is informed about disconnect
+            let client_conn_clone = connection.try_clone().unwrap();
+            let running = Arc::new(AtomicBool::new(true));
+            let _ = running.clone();
+            ctrlc::set_handler(move || {
+                net::send(&mut client_conn_clone.try_clone().unwrap(), MessageType::Quit);
```

Comment:
Deleting it leads to compiler error: `cannot borrow captured outer variable in an 'Fn' closure as mutable`